plugins/woocommerce/src/Admin/Features/MarketingRecommendations/Init.php

```diff
@@ -140,6 +161,38 @@ function ( array $plugin_data ) {
 		);
 	}
 
+	/**
+	 * Load misc recommendations from WooCommerce.com
+	 *
+	 * @since 9.5.0
+	 * @return array
+	 */
+	public static function get_misc_recommendations(): array {
+		$specs   = self::get_misc_recommendations_specs();
+		$results = self::evaluate_specs( $specs, 'misc_recommendations' );
+
+		$specs_to_return = $results['misc_recommendations'];
+		$specs_to_save   = null;
+
+		if ( empty( $specs_to_return ) ) {
+			// When misc_recommendations is empty, replace it with defaults and save for 3 hours.
+			$specs_to_save = array();
+		} elseif ( count( $results['errors'] ) > 0 ) {
+			// When misc_recommendations is not empty but has errors, save it for 3 hours.
+			$specs_to_save = $specs;
+		}
```

Comment:
What happens if there are results and no errors? Why we don't save them?

Reply:
This is where I borrowed some logic [from get_recommended_plugins() method](https://github.com/woocommerce/woocommerce/blob/9fa6e16c24542c72a35a83035cc81302a6c79e1c/plugins/woocommerce/src/Admin/Features/MarketingRecommendations/Init.php#L109). The specs [were actually saved before the code here](https://github.com/woocommerce/woocommerce/blob/e7586fdf00ad2c98ac24e03e28a4907bc04f73c3/plugins/woocommerce/src/Admin/RemoteSpecs/DataSourcePoller.php#L182-L185) when calling the WCCOM API. The logic here is if the spec is empty and has errors, we save the transient again but with fewer expired time (3 hours), so the merchant can get the new recommendations faster.